src/flashbar/collapsible-flashbar.tsx

```diff
@@ -139,16 +146,28 @@ export default function CollapsibleFlashbar({ items, ...restProps }: FlashbarPro
 
     if (initialAnimationState) {
       updateBottomSpacing();
+
+      // When collapsing with the keyboard, scroll up if necessary to avoid losing track of the focused button
+      const shouldScrollUp =
+        notificationBarRef.current &&
+        !isFlashbarStackExpanded &&
+        isElementTopBeyondViewport(notificationBarRef.current);
+
       animate({
         elements: getElementsToAnimate(),
         oldState: initialAnimationState,
         newElementInitialState: ({ top }) => ({ scale: 0.9, y: -0.2 * top }),
         onTransitionsEnd: () => setTransitioning(false),
       });
+
+      if (notificationBarRef.current && shouldScrollUp) {
```

Comment:
Is "shouldScrollUp" check really needed? If it is outside the viewport, where else could this be?

Why can you not simplify the code down to only essential part?

Reply:
Calling `scrollElementIntoViewport` on an element will scroll it to the top even if it is already inside the viewport (just verified this behavior manually on Chrome). I prefer to scroll only if it is not visible mainly because this is more similar to the behavior of the native keyboard navigation. Also, there are already quite a lot of things moving when you collapse, so I would try to not add any more visual change that is not really necessary.

Reply:
Did you test native `scrollIntoView` implementation or our helper, `scrollElementIntoView`? As I mentioned in the past comment: https://github.com/cloudscape-design/components/pull/719#discussion_r1100118357, our version has better defaults.

It was specifically created to allow you to write simpler code

Reply:
I tested the native implementation on the browser first, and then our function by unconditionally calling the it within the component. In both cases the element was scrolled to the top even if it was already present. And I couldn't see any option in the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView) that suggests it would change the behavior to not scroll if the element is already visible.

Reply:
`block: 'nearest'` is supposed to do exactly what you expect. It did it in all other places where this helper is used.

If you do not want to investigate it further, fine I can approve.

But you are also giving me a chance of creating PRs removing hundreds lines of unused code (like this one https://github.com/cloudscape-design/components/pull/709) when I actually do proper investigation instead of you

Reply:
I must have done something wrong while checking this (maybe it scrolled just because the page became shorter?):
> and then our function by unconditionally calling the it within the component

I checked this again and it didn't scroll. So I made a follow-up PR: [and then our function by unconditionally calling the it within the component](https://github.com/cloudscape-design/components/pull/734)

Reply:
Great! Appreciate a deeper dive :)